### PR TITLE
Fix OneCycleLR schedule

### LINKED ISSUE #91765: OnceCycleLR Scheduler computes wrong learning rate in last step
### 🐛 Describe the bug

The `OneCycleLR` Scheduler seems to perform a wrong last step (sudden increase back to the LR at step `total_steps-2`) when computing the learning rate. See the following code example:

```python
#!/usr/bin/env python3
import matplotlib.pyplot as plt
import torch

# Setup optimizer + scheduler
epochs = 5
steps_per_epoch = 10
params = torch.randn(10).requires_grad_(True)
optimizer = torch.optim.SGD([params], lr=0.01, momentum=0.9)
scheduler = torch.optim.lr_scheduler.OneCycleLR(
    optimizer, max_lr=0.01, steps_per_epoch=steps_per_epoch, epochs=epochs
)

# Collect lrs
lrs = []
for step in range(epochs * steps_per_epoch):
    optimizer.step()
    scheduler.step()
    lrs.append(optimizer.param_groups[0]["lr"])

# Create plot
plt.figure()
plt.title("OneCycleLR Learning Rate Scaling")
plt.plot(lrs)
plt.yscale("log")
plt.xlabel("Steps")
plt.ylabel("Learning Rate")
plt.savefig("/tmp/OneCycleLR.png")

# Print LRs
print("Min lr: ", scheduler.optimizer.param_groups[0]["min_lr"])
print("Actual last lr: ", lrs[-1])
print("Last three lrs: ", lrs[-3:])
```

**Output**:
```
$ python test_onecyclelr.py
Min lr:  4e-08
Actual last lr:  2.0168449509684565e-05
Last three lrs:  [2.0168449509684565e-05, 4e-08, 2.0168449509684565e-05]
```

OneCycleLR

### Versions

```
Collecting environment information...
PyTorch version: 1.13.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Fedora Linux 37 (Workstation Edition) (x86_64)
GCC version: (GCC) 12.2.1 20221121 (Red Hat 12.2.1-4)
Clang version: Could not collect
CMake version: version 3.25.1
Libc version: glibc-2.36

Python version: 3.10.8 (main, Nov 24 2022, 14:13:03) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-6.0.15-300.fc37.x86_64-x86_64-with-glibc2.36
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] pytorch-lightning==1.8.3.post1
[pip3] torch==1.13.0
[pip3] torchmetrics==0.11.0
[pip3] torchvision==0.14.0
[conda] numpy                     1.23.5                   pypi_0    pypi
[conda] pytorch-lightning         1.8.3.post1              pypi_0    pypi
[conda] torch                     1.13.0                   pypi_0    pypi
[conda] torchmetrics              0.11.0                   pypi_0    pypi
[conda] torchvision               0.14.0                   pypi_0    pypi
```

cc @vincentqb @jbschlosser @albanD @janeyx99

### PR DESCRIPTION
**Pull request by milesial**
Fixes #91765

Off-by-one issue: a count was divided by an index

```
Min lr:  4e-08
Actual last lr:  4e-08
Last three lrs:  [7.600093109402015e-05, 1.906643343523411e-05, 4e-08]
```

image

### CODE REVIEW

**Comment by albanD** [comment]
@milesial  why did you close this PR? It is solving the issue you raised right?

**Comment by milesial** [comment]
I commented in the issue that this might not actually be a bug. There is still a very minor issue in the ValueError message but I might open a new PR for that one.
